### DISCUSSION TITLE
Please stop expanding tabs by default

### DISCUSSION THREAD

**Discussion by dgchrt** [npm/feedback#200] [CLI] [answered]
Newly created packages produce `package*.json` files with their tabs expanded to a few spaces, which in turn won't respect editor configurations. This can be harmful to developers and should be avoided. Please refer to this article if you're new to the subject: https://alexandersandberg.com/tabs-for-accessibility/

**Comment by MylesBorins** [comment]
@diogoeichert what is you alternative suggestion? No indentations? Since many tools rely on the output of npm a change like this seems like it could be a breaking change.

If you have a specific proposal on how we should do things differently, such as respecting an exiting editor configuration, it might be a good idea to open a discussion or issue over on https://github.com/npm/rfcs

**Reply by ljharb** [reply]
a note for a future RFC, if one is written: ideally, I would expect package-lock/npm-shrinkwrap to use the same indentation as `package.json`, and ideally `npm init` should be able to respect an `.editorconfig` setting if available.

**Reply by dgchrt** [reply]
@MylesBorins quite the opposite, I'm suggesting keeping the indentation, not replacing it with spaces. This way it will work better with most tools, like you've mentioned. Thanks for the link, I'll have a look.

**Reply by dgchrt** [reply]
@ljharb the good news is that `npm` will respect the existing indentation in the files, if you fix them first. A simple `sed -i "" "s/  /\t/g" package*.json` will do.

**Comment by MylesBorins** [comment, marked as answer]
@diogoeichert if you would like to propose alternative behavior the appropriate next step would be to open an issue over at https://github.com/npm/rfcs/issues

**Reply by dgchrt** [reply]
Thanks, I've just created it: https://github.com/npm/rfcs/issues/420
